documentation/stackoverflow.workspace.adoc

```diff
@@ -3,211 +3,336 @@
+* How to use shortest path algorithms to understand relationships
 
-Throughout the guide you'll find Cypher statements that you can execute by clicking on the play button.
+In the next section, you will import the Stack Overflow dataset into Neo4j.
 
-If you want to modify a statement, click the arrow instead to copy the statement to the Cypher editor, where you can modify before executing.
+== Stake Overflow Graph Model
+[role=NX_TAB_NAV,tab=import]
+pagelaunch::[]
 
-== The Model
+The model contains data about users' questions, answers, and comments about Neo4j and Cypher.
+
+Nodes represent: 
+
+* `Users`
+* `Questions`
+* `Answers`
+* `Comments`
+* `Tags`.
 
 image::{img}/model.svg[width=100%]
 
-button::Get the StackOverflow dataset[role=NX_IMPORT_LOAD,endpoint=https://neo4j-graph-examples.github.io/stackoverflow/data/stackoverflow-data-importer.zip]
+The relationships between these nodes show:
+
+. What `User` `ASKED` a `Question`
+. What `User` `PROVIDED` an `Answer`
+. What `User` `COMMENTED` to a `Comment`
+. What `Answer` `ANSWERED` a `Question`
+. What `Comment` is `COMMENTED_ON` a `Question`
+. What `Question` is `TAGGED` with a `Tag`
+
+button::Import the StackOverflow dataset[role=NX_IMPORT_LOAD,endpoint=https://neo4j-graph-examples.github.io/stackoverflow/data/stackoverflow-data-importer.zip]
 
-=== Nodes
+Click the highlight:import/import-run-import-button[Run import] button to import the data into Neo4j.
 
-We model the `Users`, `Questions`, `Answers`, `Comments`, and `Tags` as nodes. 
+button::Explore the data[role=NX_EXPLORE_SEARCH,search=User any Question any Answer]
 
-=== Relationships
+Take a minute to _Explore_ the data:
 
-Users ask questions, comment, and provide answers.
+. Find a `Question` node
+. Follow the `ASKED` relationship to the `User` who asked the question
+. Find an `Answer` node and follow the `ANSWERED` relationship to the `Question` it answered
 
-`(:User)-[:ASKED]->(:Question)`
-`(:User)-[:COMMENTED]->(:Comment)`
-`(:User)-[:PROVIDED]->(:Answer)`
+[TIP]
+====
+You can zoom in to see more detail and double-click on nodes to expand them and see their properties.
+====
 
-Answers attempt to answer questions, comments are made on questions, and questions are associated with tags.  
+In the next section, you will use Cypher to query the graph.
 
-`(:Answer)-[:ANSWERED]->(:Question)`
-`(:Comment)-[:COMMENTED_ON]->(:Question)`
-`(:Question)-[:TAGGED]->(:Tag)`
+== Users and Questions
+[role=NX_TAB_NAV,tab=query]
+pagelaunch::[]
 
-If you want to see it yourself, run:
+Using the data model and Cypher, you can answer questions such as:
 
+* Which users asked the most questions?
+* Which users provided the most answers?
+* How are users related to each other?
+* Which questions are unanswered?
+* What answers are the most popular?
+
+The following query will use the `ASKED` relationship to find `User` and `Question` nodes.
+
+.Match User and Question nodes
 [source,cypher]
 ----
-CALL db.schema.visualization();
```

Comment:
I think keeping the schema viz is actually a good thing here, and then do the query next.

Reply:
I have added it back into the guide as a tip